pages/blog/[slug].tsx

```diff
@@ -0,0 +1,48 @@
+import { request, gql } from 'graphql-request';
+
+import { PostsMain, SlugContext, PostSlugs } from '../../index';
+import { getAllSlugs, getPost } from '../../lib/api';
+
+export default function BlogPost({ data }: PostsMain) {
+  const post = data[0];
+
+  return (
+    <div>
+      <h1>{post.title}</h1>
+      <p>
+        {post.author.name} &#8226; {post.category} &#8226;
+        {new Date(post.publishedAt).toLocaleDateString(undefined, {
+          year: 'numeric',
+          month: 'long',
+        })}
+      </p>
+      <p>{post.keywords}</p>
+      <img src={post.featuredImage.asset.url} alt={post.featuredImageAlt}></img>
+      <p>{post.body}</p>
+      <img src={post.author.image?.asset.url} alt={`${post.author.name} avatar`} />
+    </div>
+  );
+}
+
+export async function getStaticProps(context: SlugContext) {
+  const { allPost } = await request(process.env.CMS_URL as string, getPost(context.params.slug));
+
+  return {
+    props: {
+      data: allPost,
+    },
+  };
+}
+
+export async function getStaticPaths() {
+  const { allPost }: PostSlugs = await request(process.env.CMS_URL as string, getAllSlugs);
```

Comment:
Here it would be nice if you could actually move the whole request part to the `getAllSlug` function. so it would be like:
```
export const getAllSlugs = async () => {
  return request(
    process.env.CMS_URL,
    gql`
      query {
        graphql stuff
      }
   `;
}
```

and then just
```
const { allPost }: PostSlugs = await getAllSlugs();
```
That way we keep it as simple as possible in the pages stuff.